pkg/volume/csi/csi_mounter.go

```diff
@@ -426,6 +426,9 @@ func removeMountDir(plug *csiPlugin, mountPath string) error {
 	}
 	if !mnt {
 		klog.V(4).Info(log("dir not mounted, deleting it [%s]", mountPath))
+		// Deprecation: Removal of this mountPath directory and dataFile MUST be
+		// done by the CSI plugin according to the spec. This will no longer be
+		// done directly as part of TearDown by the kubelet in the future.
 		if err := os.Remove(mountPath); err != nil && !os.IsNotExist(err) {
```

Comment:
Not the datafile. Just the removal of `mountPath` should be done by the CSI driver.

Reply:
Thanks @gnufied, updated.

Reply:
One more thing is - since this function (i.e `removeMountDir` ) gets called from other places too - this may not be the right place to put the deprecation notice. How about - `TearDownAt` function- https://github.com/kubernetes/kubernetes/blob/e92dc8d9be385346534221fa67c7132a2a284a15/pkg/volume/csi/csi_mounter.go#L369 ?

Reply:
Yes, that seems a better place for it :+1: